src/fraction.ts

```diff
@@ -0,0 +1,88 @@
+import BN from "bn.js";
+
+export class Fraction {
+  private numerator: BN;
+  private denominator: BN;
+
+  constructor(numerator: BN | number, denominator: BN | number) {
+    if (typeof numerator == "number") {
+      this.numerator = new BN(numerator);
```

Comment:
I think this if statement is not necessary: 
```
> new BN(5)
<BN: 5>
> new BN(new BN(5))
<BN: 5>
```

In the second format, there is no allocation, so:
```
> a = new BN(1)
<BN: 1>
> b = new BN(2)
<BN: 2>
> c = new BN(a)
<BN: 1>
> c.iadd(b)
<BN: 3>
> a
<BN: 3>
```